### PR TITLE
docs(recipes): fixes webpack hmr config

### PR DESCRIPTION
**Pull request by delucca**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Docs
- [ ] Other... Please describe:

## What is the current behavior?

The Webpack HMR config is outdated. The guide within our docs won't work when using `nest-cli`.

Issue Number: [nestjs#9748](https://github.com/nestjs/nest/issues/9748)

## What is the new behavior?

I've updated with a config that enables HMR properly.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: content/recipes/hot-reload.md [hunk 1]
```diff
@@ -37,11 +36,10 @@ module.exports = function (options, webpack) {
     ],
     plugins: [
       ...options.plugins,
-      new webpack.HotModuleReplacementPlugin(),
       new webpack.WatchIgnorePlugin({
         paths: [/\.js$/, /\.d\.ts$/],
       }),
-      new RunScriptWebpackPlugin({ name: options.output.filename }),
+      new webpack.HotModuleReplacementPlugin(),
```
Comment (MarkPieszak) [inline content/recipes/hot-reload.md:42]: image

**Review by MarkPieszak — APPROVED** [review]
✅

File: content/recipes/hot-reload.md (hunk 1 above)
Comment (delucca) [reply content/recipes/hot-reload.md:42]: 🤣

File: content/recipes/hot-reload.md [hunk 2]
```diff
@@ -37,11 +36,10 @@ module.exports = function (options, webpack) {
     ],
     plugins: [
       ...options.plugins,
-      new webpack.HotModuleReplacementPlugin(),
       new webpack.WatchIgnorePlugin({
         paths: [/\.js$/, /\.d\.ts$/],
       }),
-      new RunScriptWebpackPlugin({ name: options.output.filename }),
```
Comment (kamilmysliwiec) [inline content/recipes/hot-reload.md:44]: Any reason why we're removing the `RunScriptWebpackPlugin`? Without it, app won't start upon successful compilation (and that's what we claim in the docs)

Comment (kamilmysliwiec) [reply content/recipes/hot-reload.md:44]: Docs suggests using `nest build` in this case though

Comment (delucca) [reply content/recipes/hot-reload.md:44]: Oh! Thanks @kamilmysliwiec 😄 

I'm not used to use HMR. I'm going to close this issue and reply to the user! Thanks for the explanation

**Pull request closed without merging**